src/Semantics.js

```diff
@@ -196,6 +196,41 @@ Semantics.prototype.checkActionDicts = function() {
   }
 };
 
+Semantics.prototype.toRecipe = function() {
+  var sb = new common.StringBuffer();
+  sb.append('(function() {\n  var grammarFn = ');
+  sb.append(this.grammar.toRecipe());
+  sb.append('  var grammar = grammarFn.call(this);\n');
+  sb.append('  var semantics = grammar.semantics()');
+
+  ['Operation', 'Attribute'].forEach(function(type) {
+    var semantics = this[type.toLowerCase() + 's'];
+    Object.keys(semantics).forEach(function(semName) {
+      var fullName = semName;
+      if (semantics[semName].formals.length > 0) {
+        fullName += '(' + semantics[semName].formals.join(', ') + ')';
+      }
+      sb.append('\n    .add' + type + '(' + JSON.stringify(fullName) + ', {');
+
+      var actions = semantics[semName].actionDict;
+      var followUp = false;
+      Object.keys(actions).forEach(function(actionName) {
+        if (actionName === '_default') {
+          return; // FIXME: check whether _default has been overwritten
```

Comment:
I thought about it. And wanted to untangle it. But at the same time I also thought that this would make it "unique" (Singleton-like) and any changes to the *default* `_default` would be made to all grammars. (Sure, functions are pseudo-immutable and usually you replace `_default` it in your semantic - if necessary at all -, but who knows...)